### ISSUE TITLE
Invalid file name

### ISSUE THREAD

**Issue by mostafaznv** [masihyeganeh/YoutubeDownloader#41]
Hi,
thanks for your nice package.

I have a problem with returned $filePath from onFinalized()
`PATH_TO_FILE\...\videos\FC_Barcelona_vs_FC_Bayern_Munich_2015_（Champions_League_semi-final_Himno,Anthem ）.3gp`

can you see spaces after ')' ?
I can't generate link to the file! browser return `404` error

original video:
https://www.youtube.com/watch?v=sMq_A0ETURQ

**Comment by mostafaznv** [comment]
oh sorry!
I didn't see it. I will work on it and if I found a better solution, I will send a pull request.

I have another request about this package!
is there any way to change cache directory? I mean we can set video and cache directory as second and third parameters of `__construct`

many thanks.

**Comment by mostafaznv** [comment]
Hi again
I implemented a custom sanitizing function for my own project.
in my function, I convert all non-english characters to the same character in english (if you set default language to `fa` or `ar`, conversion will be disabled)

feel free to change any part of my code, if you don't like it

**Comment by mostafaznv** [comment]
any updates?

**Comment by masihyeganeh** [comment]
Could you test it again with the latest version?
Now using UTF8 package.

**Comment by mostafaznv** [comment]
I tested it with latest version (2.9.1)

but unfortunately fine name generated with `）`! 
without any success

**Comment by masihyeganeh** [comment]
Please give me link to that video

**Comment by mostafaznv** [comment]
any updates?

**Comment by masihyeganeh** [comment]
<img width="873" alt="screen shot 1397-08-06 at 3 12 01 pm" src="https://user-images.githubusercontent.com/692335/47615423-d7468100-dac3-11e8-8904-82ce521991bd.png">

It is working in my PC

**Comment by mostafaznv** [comment]
![yt](https://user-images.githubusercontent.com/7619687/47616022-6fe0ff00-dacc-11e8-81fa-353227f72eac.png)

As I said [before](https://github.com/masihyeganeh/YoutubeDownloader/issues/41#issue-320746753), the problem is that ending parentheses.

please check attached image and link

**Comment by masihyeganeh** [comment]
As you can see, it is downloading the file and no error happened

**Comment by mostafaznv** [comment]
Yes I know. it downloaded for me too. but I can't generate download link for that in my website!
because file name has unknown characters for browser and browser return 404!

**Comment by masihyeganeh** [comment]
It's not the parentheses, it's the ending white space. I'll look into it

**Comment by mostafaznv** [comment]
<img width="451" alt="screen shot 1397-08-06 at 10 08 21 pm" src="https://user-images.githubusercontent.com/7619687/47620316-0a0d6b00-dafe-11e8-9c48-fc7b59b1d70c.png">
No.
It's ending parentheses but I don't know the encoding of character.
Please look at picture, It's a unit character I selected.
